### PR TITLE
Prometheus rule support

### PR DESCRIPTION
**Pull request by malcolmholmes**
Adds support for the cortex ruler, meaning we can now manage a whole mixin with Grizzly.

### CODE REVIEW

File: pkg/cortex/provider.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+type Provider struct{}
+
+// NewProvider returns a new Cortex Provider
+func NewProvider() *Provider {
+	return &Provider{}
+}
+
+// GetName returns the name of the Cortex provider
+func (p *Provider) GetName() string {
+	return "cortex"
```
Comment (Duologic) [inline pkg/cortex/provider.go:15]: This looks like a provider to talk to Hosted Prometheus, maybe name it as such?

File: testdata/cortex-rules.libsonnet [hunk 2]
```diff
@@ -0,0 +1,28 @@
+local prom = import 'prom.libsonnet';
+local promRuleGroupSet = prom.v1.ruleGroupSet;
+local promRuleGroup = prom.v1.ruleGroup;
+{
+  prometheus_metamon::
+    promRuleGroup.new('malcolm_alerts')
```
Comment (Duologic) [inline testdata/cortex-rules.libsonnet:6]:
```suggestion
    promRuleGroup.new('grizzly_alerts')
```

File: testdata/prom.libsonnet [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        name: name,
+        rules_map:: {},
+        rules_order:: [],
+        local rules_map = self.rules_map,
+        local rules_order = self.rules_order,
+        rules: [rules_map[rule] for rule in rules_order],
+      },
+
+      rule: {
+        newAlert(name, rule):: {
```
Comment (Duologic) [inline testdata/prom.libsonnet:31]:
```suggestion
      newAlertRule(name, rule):: {
```

I don't think this should be nested.

File: testdata/prom.libsonnet [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      addGroup(group):: {
+        groups_map+:: {
+          [group.name]: group,
+        },
+        groups_order+:: [group.name],
+      },
+    },
+
+    ruleGroup: {
+      new(name):: {
```
Comment (Duologic) [inline testdata/prom.libsonnet:21]:
```suggestion
    newRuleGroup(name):: {
```
Also, no need to next this.

**Comment by malcolmholmes** [comment]
I've renamed this feature from Cortex to Prometheus. These are Prometheus rules. export works with Prometheus. diff/apply/etc works with Cortex as it has an API for uploads.

File: pkg/cortex/provider.go (hunk 1 above)
Comment (malcolmholmes) [reply pkg/cortex/provider.go]: done, see below.

**Review by Duologic — APPROVED** [review]

**Pull request merged**